src/styles/calendar.scss

```diff
@@ -0,0 +1,56 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: $padding-size;
+  gap: $gap-size;
+  width: calc(7 * #{$day-size} + 6 * #{$gap-size} + 2 * #{$padding-size});
+  box-sizing: border-box;
+
+  &__day {
+    width: $day-size;
+    height: $day-size;
+    background-color: $day-color;
+    border: 1px solid black;
+    box-sizing: border-box;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    cursor: pointer;
+    transition-duration: 500ms;
+
+    &::before {
+      font-family: Arial, sans-serif;
+      font-size: 30px;
+      content: '';
+    }
+
+    &:hover {
+      background-color: $hover-color;
+      transform: translateY(-$day-hover);
+    }
+
+    @for $i from 1 through 31 {
```

Comment:
The loop generates a large number of CSS rules. Consider if there's a more efficient way to handle this, perhaps with JavaScript. If you must use CSS, ensure this is the best approach for performance.